### PR TITLE
Add classes consistency check

### PR DESCRIPTION
**Pull request by sovrasov**
In case of evaluation on dataset which has a fewer amount of classes than model knows, labelmap will be built for correct matching.

### CODE REVIEW

File: torchreid/models/common.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def __init__(self,
           self.classification = classification
           self.contrastive = contrastive
           self.pretrained = pretrained
+          self.classification_classes = {}
```
Comment (evgeny-izutov) [inline torchreid/models/common.py:31]: Can't find any initialization code for this field.

Comment (sovrasov) [reply torchreid/models/common.py:31]: See `load_pretrained_weights`

Comment (sovrasov) [reply torchreid/models/common.py:31]: Classes on evaluation are checked only if we have a reference in checkpoint.

Comment (evgeny-izutov) [reply torchreid/models/common.py:31]: But who is responsible to pass `classes_map` into `checkpoint`?

Comment (sovrasov) [reply torchreid/models/common.py:31]: `Engine.save_model` will do this if possible

Comment (evgeny-izutov) [reply torchreid/models/common.py:31]: I see, Ok.

**Comment by evgeny-izutov** [comment]
@sovrasov Is this PR ready to merge?

**Comment by sovrasov** [comment]
I think so. Thanks for comments, I've found 2 more potential bugs.

**Review by evgeny-izutov — APPROVED** [review]

**Pull request merged**